pkg/queue/concurrency_state.go

```diff
@@ -47,6 +62,10 @@ func ConcurrencyStateHandler(logger *zap.SugaredLogger, h http.Handler, pause, r
 		}
 	}()
 
+	retryChannel := make(chan RetryElement, 1)
```

Comment:
One thing to be aware of: there's a PR open to rewrite the ConcurrencyStateHandler from a channel based approach to an atomic/mux based one ([PR 11909](https://github.com/knative/serving/pull/11909)). That's _probably_ the direction we'll be moving in, so may want to start thinking along those lines.